src/bundler/bundle_v2.zig

```diff
@@ -9389,6 +9484,163 @@ pub const LinkerContext = struct {
         }
     }
 
+    fn generateCompileResultForHTMLChunkImpl(worker: *ThreadPool.Worker, c: *LinkerContext, chunk: *Chunk, chunks: []Chunk) CompileResult {
+        const parse_graph = c.parse_graph;
+        const input_files = parse_graph.input_files.slice();
+        const sources = input_files.items(.source);
+        const import_records = c.graph.ast.items(.import_records);
+
+        // We want to rewrite the HTML with the following transforms:
+        // 1. Remove all <script> and <link> tags which were not marked as
+        //    external. This is defined by the source_index on the ImportRecord,
+        //    when it's not Index.invalid then we update it accordingly. This will
+        //    need to be a reference to the chunk or asset.
+        // 2. For all other non-external URLs, update the "src" or "href"
+        //    attribute to point to the asset's unique key. Later, when joining
+        //    chunks, we will rewrite these to their final URL or pathname,
+        //    including the public_path.
+        // 3. If a JavaScript chunk exists, add a <script type="module" crossorigin> tag that contains
+        //    the JavaScript for the entry point which uses the "src" attribute
+        //    to point to the JavaScript chunk's unique key.
+        // 4. If a CSS chunk exists, add a <link rel="stylesheet" href="..." crossorigin> tag that contains
+        //    the CSS for the entry point which uses the "href" attribute to point to the
+        //    CSS chunk's unique key.
+        // 5. For each imported module or chunk within the JavaScript code, add
+        //    a <link rel="modulepreload" href="..." crossorigin> tag that
+        //    points to the module or chunk's unique key so that we tell the
+        //    browser to preload the user's code.
+        // We might also want to force <!DOCTYPE html> at the top of the file, but I'm not sure about that yet.
```

Comment:
its fine to take no action re:doctype

the one person who wants to bundle quirks mode html can do so